app/controllers/mccloud/AddAnotherPensionSchemeController.scala

```diff
@@ -0,0 +1,135 @@
+            returnUrl = controllers.routes.ReturnToSchemeDetailsController.returnToSchemeDetails(srn, startDate, accessType, version).url,
+            schemeName = schemeName
+          )
+
+          val preparedForm = request.userAnswers.get(AddAnotherPensionSchemePage(chargeType, index, schemeIndex)) match {
+            case None        => form(chargeTypeDescription)
+            case Some(value) => form(chargeTypeDescription).fill(value)
+          }
+
+          val json = Json.obj(
+            "srn" -> srn,
+            "startDate" -> Some(localDateToString(startDate)),
+            "form" -> preparedForm,
+            "viewModel" -> viewModel,
+            "radios" -> Radios.yesNo(preparedForm("value"))
+          )
+          renderer.render("mccloud/addAnotherPensionScheme.njk", json).map(Ok(_))
+
+        }
+    }
+
+  private def form(memberName: String)(implicit messages: Messages): Form[Boolean] =
+    formProvider(messages("addAnotherPensionScheme.error.required", memberName))
+
+  def onSubmit(chargeType: ChargeType,
+               mode: Mode,
+               srn: String,
+               startDate: LocalDate,
+               accessType: AccessType,
+               version: Int,
+               index: Index,
+               schemeIndex: Index): Action[AnyContent] =
+    (identify andThen getData(srn, startDate) andThen requireData).async { implicit request =>
+      DataRetrievals.retrieveSchemeName { schemeName =>
+        val chargeTypeDescription = Messages(s"chargeType.description.${chargeType.toString}")
+        form(chargeTypeDescription)
+          .bindFromRequest()
+          .fold(
+            formWithErrors => {
+              val viewModel = GenericViewModel(
+                submitUrl = routes.AddAnotherPensionSchemeController.onSubmit(chargeType, mode, srn, startDate, accessType, version, index, schemeIndex).url,
+                returnUrl = controllers.routes.ReturnToSchemeDetailsController.returnToSchemeDetails(srn, startDate, accessType, version).url,
+                schemeName = schemeName
+              )
+
+              val json = Json.obj(
+                "srn" -> srn,
+                "startDate" -> Some(localDateToString(startDate)),
+                "form" -> formWithErrors,
+                "viewModel" -> viewModel,
+                "radios" -> Radios.yesNo(formWithErrors("value"))
+              )
+              renderer.render("mccloud/addAnotherPensionScheme.njk", json).map(BadRequest(_))
+            },
+            value =>
+              for {
+                updatedAnswers <- Future.fromTry(userAnswersService.set(AddAnotherPensionSchemePage(chargeType, index, schemeIndex), value, mode))
+                _ <- userAnswersCacheConnector
+                  .savePartial(request.internalId, updatedAnswers.data, chargeType = Some(chargeType), memberNo = Some(index.id))
```

Comment:
I don't think you need to save this. It is only used for navigation

Reply:
removed unwanted code